grails-app/services/org/broadinstitute/orsp/PermissionService.groovy

```diff
@@ -10,31 +10,61 @@ class PermissionService implements UserInfo {
     UserService userService
 
     // get issue's collaborators as a List<String>
-    List<String> getIssueCollaborators(Issue issue) {
-      issue.extraProperties.findAll ({ it.name == IssueExtraProperty.COLLABORATOR }).collect { property -> property.value }
+    List<String> getIssueCollaborators(Map<String, List<String>> extraProperties) {
+        List<String> results = new ArrayList<String>()
+
+        def collaboratorsValues = extraProperties.findAll ({ it.key == IssueExtraProperty.COLLABORATOR }).values()
+        if (collaboratorsValues.size() != 0) {
+            results = collaboratorsValues.first()
+        }
+
+        collaboratorsValues = extraProperties.findAll ({ it.key == IssueExtraProperty.COLLABORATORS }).values()
+        if (results.size() == 0 && collaboratorsValues.size() != 0) {
+            results = collaboratorsValues.first()
+        }
+
+        results
```

Comment:
This whole method is a little confusing. 
* Why aren't the collaborators *all* of the collaborators? 
* Why is it only the first one (in each of the two current cases)? 
* Why do we mutate the results? 
* Why iterate twice over the `extraProperties`? 

It would make more sense to me to have something like this:
```
List<String> getIssueCollaborators(Map<String, List<String>> extraProperties) {
    extraProperties.
        findAll({ [IssueExtraProperty.COLLABORATOR, IssueExtraProperty.COLLABORATORS].contains(it.key) }).
        values()
}
```

Reply:
It's not the first of the values but the result of this is a array with the Collaborators user names within an array. I need to take out from there the actual values. This suggestion is working but I still need use `.first()`

Reply:
If that's true, then there's something wrong with how you're constructing the values inside `collaboratorsValues`. It should just be just a simple list of names. If instead it is a list of lists, then the underlying method should be super clear about what it's returning.

Reply:
I not agree, `Issue`s are being instantiated in some way that the `extraProperties` are in the same way as my DTO. But I'll fix boths.